src/webgl/p5.RendererGL.js

```diff
@@ -32,85 +34,114 @@ defineStrokeJoinEnum('ROUND', 0);
 defineStrokeJoinEnum('MITER', 1);
 defineStrokeJoinEnum('BEVEL', 2);
 
-const lightingShader = readFileSync(
+const getCachedShader= (shaderKey, shaderSource)=>{
+  if(!shaderCache[shaderKey]){
+    shaderCache[shaderKey]=p5.createShader(shaderSource);
```

Comment:
Shaders also have both a vertex and a fragment shader, not a single shader source, so you'd need to add two parameters to the function and pass both into `createShader`.

That said, your `_getCubemapShader` implementation on its own does all the caching we need, so we might not need this in addition to that.